File: common/src/main/java/generations/gg/generations/core/generationscore/common/world/level/block/FlabebeFlowerBlock.java
package generations.gg.generations.core.generationscore.common.world.level.block;

import net.minecraft.core.BlockPos;
import net.minecraft.world.level.BlockGetter;
import net.minecraft.world.level.block.BushBlock;
import net.minecraft.world.level.block.state.BlockState;
import net.minecraft.world.phys.shapes.CollisionContext;
import net.minecraft.world.phys.shapes.Shapes;
import net.minecraft.world.phys.shapes.VoxelShape;
import org.jetbrains.annotations.NotNull;

public class FlabebeFlowerBlock extends BushBlock {
    private final VoxelShape SHAPE = Shapes.box(0.1, 0.0D, 0.1, 0.9, 0.8, 0.9);

    public FlabebeFlowerBlock(Properties properties) {
        super(properties);
    }

    @Override
    public @NotNull VoxelShape getShape(@NotNull BlockState state, @NotNull BlockGetter level, @NotNull BlockPos pos, @NotNull CollisionContext context) {
        return SHAPE;
    }
}

File: common/src/main/java/generations/gg/generations/core/generationscore/common/world/level/block/FlabebeFlowerBlock.kt
package generations.gg.generations.core.generationscore.common.world.level.block

import com.mojang.serialization.MapCodec
import net.minecraft.core.BlockPos
import net.minecraft.world.level.BlockGetter
import net.minecraft.world.level.block.BushBlock
import net.minecraft.world.level.block.state.BlockState
import net.minecraft.world.phys.shapes.CollisionContext
import net.minecraft.world.phys.shapes.Shapes
import net.minecraft.world.phys.shapes.VoxelShape

class FlabebeFlowerBlock(properties: Properties) : BushBlock(properties) {
    override fun codec(): MapCodec<FlabebeFlowerBlock> = CODEC

    public override fun getShape(
        state: BlockState,
        level: BlockGetter,
        pos: BlockPos,
        context: CollisionContext
    ): VoxelShape {
        return SHAPE
    }

    companion object {
        private val SHAPE: VoxelShape = Shapes.box(0.1, 0.0, 0.1, 0.9, 0.8, 0.9)
        val CODEC = simpleCodec(::FlabebeFlowerBlock)
    }
}
